src/icesat2_tracks/ICEsat2_SI_tools/spectral_estimates.py

```diff
@@ -394,126 +405,128 @@ def __init__(self, x_grid, data, Lpoints, ov=None, window=None):
             other arributes are in the .attr dict.
         """
 
-        self.Lpoints  = Lpoints
-        self.ov       = int(Lpoints/2) if ov is None else ov #when not defined in create_chunk_boundaries then L/2
+        self.Lpoints = Lpoints
+        self.ov = int(Lpoints / 2) if ov is None else ov
 
-        self.data     = data
+        self.data = data
 
         # create subsample k
-        self.k, self.dk = calc_freq_fft(x_grid, Lpoints) # return 1/ unit of frid points
+        self.k, self.dk = calc_freq_fft(x_grid, Lpoints)
         # to get the waveumber units (2  pi/ lambda), multiply by 2 pi
-        self.k, self.dk = self.k * 2 * np.pi, self.dk  * 2 * np.pi 
+        self.k, self.dk = self.k * 2 * np.pi, self.dk * 2 * np.pi
 
-        # print(self.k[0], self.k[-1])
-        # print(self.dk)
         # create window
-        self.win    = create_window(Lpoints)
+        self.win = create_window(Lpoints)
 
     def cal_spectrogram(self, data=None, name=None):
-
         """
         defines apply function and calculated all sub-sample sprectra using map
         """
-        import xarray as xr
 
         DATA = self.data if data is None else data
         Lpoints, dk = self.Lpoints, self.dk
-        win =self.win
+        win = self.win
 
         def calc_spectrum_apply(stancil):
 
             "returns spectrum per stencil, detrends and windows the data"
-            from scipy.signal import detrend
-
-            idata = DATA[stancil[0]:stancil[-1]]
+            idata = DATA[stancil[0] : stancil[-1]]
             idata = detrend(idata) * win
 
-            return stancil[1], calc_spectrum_fft(idata , dk, Lpoints)
-
-        # def test_func(i_stancil):
-        #     return i_stancil[1], yy[i_stancil[0]:i_stancil[-1]].shape
+            return stancil[1], calc_spectrum_fft(idata, dk, Lpoints)
 
-        # %% derive L2 stancil
-        stancil_iter = create_chunk_boundaries(Lpoints, DATA.size, ov= self.ov)
+        # derive L2 stancil
+        stancil_iter = create_chunk_boundaries(Lpoints, DATA.size, ov=self.ov)
         # apply func to all stancils
-        D_specs = dict(map(calc_spectrum_apply,stancil_iter))
+        D_specs = dict(map(calc_spectrum_apply, stancil_iter))
 
         chunk_positions = np.array(list(D_specs.keys()))
-        self.N_stancils = len(chunk_positions) # number of spectal relazations
 
-        # repack data, create xarray
-        self.spec_name = 'power_spec' if name is None else name
-        G =dict()
-        for xi,I in D_specs.items():
-            G[xi] = xr.DataArray(I,  dims=['k'], coords={'k': self.k, 'x': xi } , name=self.spec_name)
+        # number of spectal relazations
+        self.N_stancils = len(chunk_positions)
 
-        self.G = xr.concat(G.values(), dim='x').T#.to_dataset()
+        # repack data, create xarray
+        self.spec_name = "power_spec" if name is None else name
```

Comment:
If the previous suggestion is added, this line can be deleted.
```suggestion
        self.spec_name = name
```